spec/index.adoc

```diff
@@ -1100,6 +1100,15 @@ Status `1` indicates running, `2` indicates stopping, and `3` indicates stopped.
 +
 Status `3` (stopped) can be observed, for example, in `canister_pre_upgrade` and can be used to prevent accidentally upgrading a canister that is not fully stopped.
 
+[#system-api-canister-state-counter]
+=== Canister state counter
+
+For each canister, the system maintains a _state counter_. Upon canister creation, it is set to 0, and it is guaranteed to be incremented upon every change to the canister's state (except for changes affecting only the canister's time), i.e., upon every (successful and unsuccessful) message execution (ingress message, inter-canister call, callback execution) on that canister, successful management canister call (`update_settings`, `install_code`, `uninstall_code`, `stop_canister`, `start_canister`, `deposit_cycles`, and `provisional_top_up_canister`) on that canister, change of the canister status of that canister from stopping to stopped, code uninstallation due to that canister running out of cycles, successful `inspect_message` execution for ingress messages, and charging canisters for (compute and memory) resource allocation and usage. The system can arbitrarily increment the state counter also if the canister's state does not change.
```

Comment:
> change of the canister status of that canister from stopping to stopped

What about change of canister status from running to stopping? Or stopped to running? You are incrementing the counter in the formal model section in these cases. More generally, are you trying to capture every possible case where a change in the canister state might happen or is this list supposed to be more like examples?

Reply:
The cases you describe are covered by "successful IC management canister calls". Going from stopping to stopped is an internal system transition not associated to a particular IC management canister call.